gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -766,6 +767,7 @@ protected void setup(Context context) {
       for (Map.Entry<String, ConfigValue> entry : dynamicConfig.entrySet()) {
         this.jobState.setProp(entry.getKey(), entry.getValue().unwrapped().toString());
         configuration.set(entry.getKey(), entry.getValue().unwrapped().toString());
+        gobblinJobState.setProp(entry.getKey(), entry.getValue().unwrapped().toString());
```

Comment:
Is there anyway to only keep `gobblinJobState` here, instead of have `configuration` object and `gobblinJobState`  ?

Reply:
I did consider it. I think this needs a significant clean up. In addition to configuration and gobblinJobState, we also have jobProps (which is a java.util.Properties). It seems that there is overlap in configs across these types, and it is unclear which property should go where. Arguably, since Hadoop APIs use Configuration, we should set all properties in the configuration object. But I am open to suggestions. I will create a tech-debt ticket and use that for addressing this.

Reply:
+1 Sounds good.